packages/wallets/src/metamask/metamask.page.ts

```diff
@@ -114,11 +114,24 @@ export class MetamaskPage implements WalletPage {
       if (await this.page.getByText('Not right now').isVisible())
         await this.page.click('text=Not right now');
 
-      const gotItBtn = await this.page.getByText('Got it');
+      const gotItBtn = this.page.getByText('Got it');
       if (await gotItBtn.first().isVisible()) await gotItBtn.first().click();
+
+      // reject tx in queue.
+      while (await this.isTxInQueue()) await this.rejectTx(this.page);
     });
   }
 
+  async isTxInQueue() {
+    try {
+      const rejectTxBtn = this.page.getByTestId('page-container-footer-cancel');
+      await rejectTxBtn.waitFor({ state: 'visible', timeout: 1000 });
```

Comment:
One time I caught the case when this waitFor return `true`, but that was the previous button
Then test failed cause the `rejectTx()` didn't find the Reject button